File: examples/java/kora-java-s3-client-minio/src/main/java/io/koraframework/example/s3/minio/RootService.java
package io.koraframework.example.s3.minio;

import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Root;

@Root
@Component
public class RootService {

    private final SyncS3Client syncS3Client;

    public RootService(SyncS3Client syncS3Client) {
        this.syncS3Client = syncS3Client;
    }
}

File: examples/kotlin/kora-kotlin-s3-client-minio/src/main/kotlin/io/koraframework/kotlin/example/s3/minio/RootService.kt
package io.koraframework.kotlin.example.s3.minio

import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Root
import io.koraframework.s3.client.annotation.S3
import io.koraframework.s3.client.model.*

@Root
@Component
class RootService(private val syncS3Client: SyncS3Client)

